src/crimson/mgr/client.cc

```diff
@@ -58,31 +58,47 @@ seastar::future<> Client::ms_dispatch(crimson::net::Connection* conn,
   }
 }
 
+seastar::future<> Client::ms_handle_connect(crimson::net::ConnectionRef c)
+{
+  if (conn == c) {
+    // ask for the mgrconfigure message
+    auto m = ceph::make_message<MMgrOpen>();
+    m->daemon_name = local_conf()->name.get_id();
+    return conn->send(std::move(m));
```

Comment:
> send the MMgrOpen message in `ms_handle_connect()` to make it explicit that we sends this message when the connection is established.

Well, I tend to think the original is better, because this version won't guarantee that `MMgrOpen` will be the first message to send once connected. `report_timer` may be invoked to send `pg_stats` during handshake, which will be queued before `MMgrOpen`.

But I'm good if we don't have this constraint.

Reply:
could you reference https://github.com/ceph/ceph/pull/33797#discussion_r390095871 for the reason?